### PR TITLE
Fix internal thinktime defaults

### PR DESCRIPTION
**Pull request by atluq**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [x] documentation updated

**Squash commit message**

Fix internal thinktime GUI defaults

- Marshal default values if zero value in json
- Set smartsearch default to static delay 1s for less confusion
- Change action internal thinktimes to values, no null check needed because zero value is already checked.
- Add GUIDefaults interface

**Info**

Optional informative text (not to be included in commit message) relevant to reviewers.

### CODE REVIEW

File: scenario/smartsearch.go [hunk 1]
```diff
@@ -167,6 +165,13 @@ func (settings *SmartSearchSettings) UnmarshalJSON(bytes []byte) error {
 	}
 	return nil
 }
+func (settings SmartSearchSettings) MarshalJSON() ([]byte, error) {
+	settings.SelectionThinkTime = thinkTimeWithFallback(
+		settings.SelectionThinkTime,
+		smartSearchDefaultThinktimeSettings,
+	)
+	return json.Marshal(settings)
+}
```
Comment (DnlLrssn) [inline scenario/smartsearch.go:174]: infinite loop

File: docs/settingup.md [hunk 2]
```diff
@@ -432,7 +432,7 @@ Perform a query in the Qlik Sense hub insight advisor.
 * `app`: Optional name of app to pick in followup queries. If not set, a random app is picked.
 * `saveimages`: Save images of charts to file.
 * `saveimagefile`: File name of saved images. Defaults to server side file name. Supports [Session Variables](https://github.com/qlik-trial/gopherciser-oss/blob/master/docs/settingup.md#session-variables).
-* `thinktime`: Settings for the `thinktime` action, which is automatically inserted before each followup.
+* `thinktime`: Settings for the `thinktime` action, which is automatically inserted before each followup. Defaults to a uniform distribution with mean=8 and deviation=4.
```
Comment (SeBBBe) [inline docs/settingup.md:435]: Wrong!

**Review by SeBBBe — CHANGES_REQUESTED** [review]

File: scenario/smartsearch.go (hunk 1 above)
Comment (atluq) [reply scenario/smartsearch.go]: 👍

File: docs/settingup.md (hunk 2 above)
Comment (atluq) [reply docs/settingup.md:435]: Note for hubadvisor. https://github.com/qlik-oss/gopherciser/blob/fb58db4096402ae78fb7f1980ff40f1828ebe99a/scenario/askhubadvisor.go#L60

I kept this to not change results of old scripts with implicit thinktime.

**Review by DnlLrssn — APPROVED** [review]

**Review by SeBBBe — APPROVED** [review]

**Pull request merged**